include/pybind11/docstring_options.h

```diff
@@ -0,0 +1,73 @@
+/*
+    pybind11/docstring_options.h: selectively suppress the automatic generation of docstrings and function signatures.
+
+    Copyright (c) 2016 Wenzel Jakob <wenzel.jakob@epfl.ch>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include "common.h"
+
+NAMESPACE_BEGIN(pybind11)
+
+class docstring_options
+{
+  public:
+
+    struct state {
+        bool show_user_defined; //< Include user-defined texts in docstrings.
+        bool show_signatures;   //< Inclcude auto-generated function signatures in docstrings.
+    };
+
+    PYBIND11_NOINLINE inline static state &global_state() {
+        static state instance = { true, true };
+        return instance;
+    }
+
+    docstring_options(bool show_all)
```

Comment:
This constructor needs an "explicit" (to prevent implicit conversion from bool).